bin/inx-chronicle/src/listener.rs

```diff
@@ -1,75 +1,198 @@
 // Copyright 2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-//! The [`InxListener`] subscribes to events from INX and forwards them as an [`InxEvent`] via a Tokio unbounded
+//! The [`InxListener`] subscribes to events from INX and forwards them via a Tokio unbounded
 //! channel.
 
-use futures::StreamExt;
+use std::{fmt::Debug, marker::PhantomData};
+
+use async_trait::async_trait;
+use chronicle::{
+    inx::{InxConfig, InxError},
+    runtime::{
+        actor::{context::ActorContext, envelope::HandleEvent, error::ActorError, handle::Addr, report::Report, Actor},
+        config::ConfigureActor,
+        error::RuntimeError,
+    },
+};
 use inx::{
     client::InxClient,
     proto::{MessageFilter, NoParams},
-    Channel,
+    Channel, Status,
 };
 use log::info;
-use tokio::task::JoinHandle;
+use thiserror::Error;
 
-use crate::broker::BrokerAddr;
+use crate::broker::Broker;
 
-#[derive(Debug)]
-pub enum InxEvent {
-    Message(inx::proto::Message),
-    LatestMilestone(inx::proto::Milestone),
+#[derive(Debug, Error)]
+pub enum InxListenerError {
+    #[error("The broker actor is not running")]
+    MissingBroker,
+    #[error(transparent)]
+    Inx(#[from] InxError),
+    #[error(transparent)]
+    Read(#[from] Status),
+    #[error(transparent)]
+    Runtime(#[from] RuntimeError),
 }
 
+#[derive(Debug)]
 pub struct InxListener {
-    // TODO: Consider storing `inx_client` to potentially restart some streams.
-    message_handle: JoinHandle<()>,
-    latest_milestone_handle: JoinHandle<()>,
+    config: InxConfig,
+    broker_handle: Addr<Broker>,
```

Comment:
Since you named this field `broker_handle` and not `broker_addr` maybe consider renaming `Addr` to `Handle`.

Reply:
Hey it wasn't even my idea to call it `Addr`.

Reply:
`broker_addr` is fine :trollface: